models/api.article.model.js

```diff
@@ -0,0 +1,19 @@
+const db = require("../db/connection");
+
+exports.fetchArticle = (id) => {
+  if (isNaN(id)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Error - bad request, invalid article_id",
+    });
+  }
+  const sqlQuery = `SELECT * FROM articles WHERE article_id = $1 `;
+  const queryValues = [id];
+
+  return db.query(sqlQuery, queryValues).then(({ rows }) => {
+    if (rows.length === 0) {
+      throw { status: 404, msg: "Error - article not found" };
```

Comment:
This is the right behaviour and valid code (I have learned!). I'm interested why you've chosen `throw` over the way you did it above with `promise.reject`.